ksqldb-parser/src/main/java/io/confluent/ksql/parser/SyntaxErrorValidator.java

```diff
@@ -127,16 +137,62 @@ public void syntaxError(
       return;
     }
 
-    if (offendingSymbol instanceof Token && isKeywordError(
```

Comment:
We can probably simplify this method a bit. Here is my take, but feel free to change the way you like:

```java
    final String errorMessage;
    if (offendingSymbol instanceof Token) {
      final StringBuilder sb = new StringBuilder("Syntax Error\n");
      final String token = ((Token) offendingSymbol).getText();
      if (isKeywordError(message, token)) {
        sb.append(
            String.format("\"%s\" is a reserved keyword and it can't be used as an identifier."
                + " You can use it as an identifier by escaping it as '%s' ",
                token, token));
      } else if (e instanceof InputMismatchException) {
        sb.append(
            getExpectations(message,  e.getOffendingToken().getText(), line, charPositionInLine)
        );
      } else {
        sb.append(message);
      }
      errorMessage = sb.toString();
    } else {
      errorMessage = message;
    }

    throw new ParsingException(errorMessage, e, line, charPositionInLine);
  }
```

I find this version easier to understand because we separate the logic of creating an error message from throwing it.